File: src/main/java/net/orcinus/galosphere/util/GByteBufCodecs.java
package net.orcinus.galosphere.util;

import io.netty.buffer.ByteBuf;
import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.network.codec.StreamCodec;

import java.util.UUID;

public class GByteBufCodecs {
    public static final StreamCodec<ByteBuf, UUID> UUID = new StreamCodec<>() {
        @Override
        public java.util.UUID decode(ByteBuf object) {
            return FriendlyByteBuf.readUUID(object);
        }

        @Override
        public void encode(ByteBuf object, java.util.UUID object2) {
            FriendlyByteBuf.writeUUID(object, object2);
        }
    };
}

File: src/main/java/dissonance/util/GByteBufCodecs.kt
package dissonance.util

import io.netty.buffer.ByteBuf
import net.minecraft.network.FriendlyByteBuf
import net.minecraft.network.codec.StreamCodec
import java.util.UUID

object GByteBufCodecs
{
	val UUID = object : StreamCodec<ByteBuf, UUID>
	{
		override fun decode(ob: ByteBuf) = FriendlyByteBuf.readUUID(ob)
		override fun encode(ob: ByteBuf, object2: UUID) = FriendlyByteBuf.writeUUID(ob, object2)
	}
}